receiver/tlscheckreceiver/scraper.go

```diff
@@ -5,27 +5,91 @@ package tlscheckreceiver // import "github.com/open-telemetry/opentelemetry-coll
 
 import (
 	"context"
+	"crypto/tls"
+	"errors"
+	"sync"
+	"time"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/receiver"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver/internal/metadata"
 )
 
+var (
+	ErrMissingTargets = errors.New(`No targets specified`)
+)
+
 type scraper struct {
-	// include string
-	logger *zap.Logger
-	mb     *metadata.MetricsBuilder
+	cfg                *Config
+	settings           component.TelemetrySettings
+	mb                 *metadata.MetricsBuilder
+	getConnectionState func(endpoint string) (tls.ConnectionState, error)
+}
+
+func getConnectionState(endpoint string) (tls.ConnectionState, error) {
+	conn, err := tls.Dial("tcp", endpoint, &tls.Config{InsecureSkipVerify: true})
+	if err != nil {
+		return tls.ConnectionState{}, err
+	}
+	defer conn.Close()
+	return conn.ConnectionState(), nil
+}
+
+func (s *scraper) scrapeEndpoint(endpoint string, wg *sync.WaitGroup, mux *sync.Mutex) {
+	defer wg.Done()
+
+	mux.Lock()
```

Comment:
What is the mutex meant to protect? If it's only intended to protect the metricbuilder you should not lock until the last moment. As it stands this is written as if its in parallel but the mutex does not allow any parallelization.